sdk/spring/spring-cloud-azure-autoconfigure/src/main/java/com/azure/spring/cloud/autoconfigure/jms/ServiceBusJmsPasswordlessConfiguration.java

```diff
@@ -0,0 +1,46 @@
+// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License.
+
+package com.azure.spring.cloud.autoconfigure.jms;
+
+import com.azure.spring.cloud.autoconfigure.context.AzureGlobalProperties;
+import com.azure.spring.cloud.autoconfigure.jms.properties.AzureServiceBusJmsProperties;
+import com.azure.spring.cloud.core.implementation.util.AzurePasswordlessPropertiesUtils;
+import org.apache.qpid.jms.JmsConnectionExtensions;
+import org.springframework.boot.autoconfigure.EnableAutoConfiguration;
+import org.springframework.boot.autoconfigure.condition.ConditionalOnMissingBean;
+import org.springframework.boot.autoconfigure.condition.ConditionalOnProperty;
+import org.springframework.context.annotation.Bean;
+import org.springframework.context.annotation.Configuration;
+
+/**
+ * {@link EnableAutoConfiguration Auto-configuration} for Azure Service Bus JMS passwordless support.
+ *
+ * @since 4.7.0
+ */
+@Configuration(proxyBeanMethods = false)
+@ConditionalOnProperty(value = "spring.jms.servicebus.passwordless-enabled", havingValue = "true")
+public class ServiceBusJmsPasswordlessConfiguration {
```

Comment:
We can make this class package-private.

Reply:
ok, but how about other existing Configuration classes.

Reply:
This only applies to newly added configuration classes, because we want to reduce the public API surface, without bringing any breaking changes.